roombapy/remote_client.py

```diff
@@ -2,13 +2,23 @@
 import logging
 import ssl
 from functools import cache
+from typing import Any, Callable
 
 import paho.mqtt.client as mqtt
 
-from roombapy.const import MQTT_ERROR_MESSAGES
+from roombapy.const import MQTT_ERROR_MESSAGES, TransportErrorMessage
 
 MAX_CONNECTION_RETRIES = 3
 
+ConnectCallback = Callable[[TransportErrorMessage], None]
```

Comment:
```suggestion
ConnectionCallback = Callable[[TransportErrorMessage], None]
```
Maybe this is a more correct name